src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,297 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InlineMethod
+{
+    internal abstract partial class AbstractInlineMethodRefactoringProvider<TInvocationSyntaxNode, TExpressionSyntax, TArgumentSyntax>
+        where TInvocationSyntaxNode : SyntaxNode
+        where TExpressionSyntax : SyntaxNode
+        where TArgumentSyntax : SyntaxNode
+    {
+        /// <summary>
+        /// Information about the callee method parameters to compute <see cref="InlineMethodContext"/>.
+        /// </summary>
+        private class MethodParametersInfo
+        {
+            /// <summary>
+            /// Parameters map to identifier argument.
+            /// </summary>
+            public ImmutableArray<(IParameterSymbol parameterSymbol, SyntaxNode identifierSyntaxNode)> ParametersWithIdentifierArgument { get; }
+
+            /// <summary>
+            /// Parameters map to variable declaration argument (out var declaration in C#)
+            /// </summary>
+            public ImmutableArray<(IParameterSymbol parameterSymbol, SyntaxNode variableDeclarationNode)> ParametersWithVariableDeclarationArgument { get; }
+
+            /// <summary>
+            /// Parameters map to other expression arguments. Note: params array could maps to multiple arguments.
+            /// </summary>
+            public ImmutableArray<(IParameterSymbol parameterSymbol, ImmutableArray<SyntaxNode> arguments)> ParametersNeedGenerateDeclarations { get; }
```

Comment:
the name of the member doesn't match the comment.  what does "expression argumetns" have to do with "NeedGeenrateDeclarations".  Needs comments.